x-pack/plugins/maps/server/plugin.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from 'src/core/server';
+import { PluginSetupContract as FeaturesPluginSetupContract } from '../../features/server';
+
+interface SetupDeps {
+  features: FeaturesPluginSetupContract;
+}
+
+export class MapsPlugin implements Plugin {
+  constructor(initializerContext: PluginInitializerContext) {}
+  setup(core: CoreSetup, plugins: SetupDeps) {
+    plugins.features.registerFeature({
+      id: 'maps',
+      name: i18n.translate('xpack.maps.featureRegistry.mapsFeatureName', {
+        defaultMessage: 'Maps',
+      }),
+      order: 600,
+      icon: 'gisApp',
```

Comment:
If moved constants to KP, I'd have had to update all the references in LP.
If I kept them in LP, I'd has had to import legacy code in KP, which we want to avoid. 
I duplicated them in both KP & LP, it have defeated the purpose to be the only source of truth.

Reply:
> If moved constants to KP, I'd have had to update all the references in LP.

They were already moved to NP. the legacy plugin was importing them from the KP plugin.

I'm personally +1 with you regarding the 'value' of such constants, but the decision of keeping them or not should still probably remains to the owners.

Unless for technical reasons such as

> I’m not sure dynamic nature of ${ROUTE}/${TYPE}/endpoint is compatible with tools generating API spec that we want to add later.

or course.

> I think Maps team will update code later as they want

SGTM

Reply:
switched back to constants